src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Routing/Encoded/EncodedConstrainedCatchAll.razor

```diff
@@ -0,0 +1,24 @@
+﻿@page "/🙂/routing/constrained-catch-all/{*parameter:regex(http:%2F%2Fwww\\.example\\.com%2Flogin%2Fcallback\\/another)}"
```

Comment:
![My eyes](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExYmFlcWM2aTRyMmowejJlaTZjYTN1empjOXh5YnU2OTBxOXc5YjVsNSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/oS4bJ2sRXKVDa/giphy.gif)

Reply:
Yeah, this is a bit extreme, but I wanted to make sure we cover all the edge cases correctly, as this is very tricky.